page.js

```diff
@@ -145,16 +146,33 @@ class Page {
     return this.validate(newState, this.stateSchema)
   }
 
+  get lang () {
+    if(!this.__lang) {
+      this.lang = 'en'
+    }
+    return this.__lang
+  }
+
+  set lang (lang) {
+    if (lang) {
+      this.__lang = lang
+    }
+  }
+
+  setLangFromRequest (request) {
+    this.lang = request.query.lang || request.yar.get('lang') || 'en'
+    if (this.lang !== request.yar.get('lang')) {
+      request.i18n.setLocale(this.lang)
+      request.yar.set('lang', this.lang)
+    }
+  }
```

Comment:
Is `this` unique per request or shared?

Reply:
My guess is it's not, so we shouldn't store request-specific data on it - can you replace `this.lang` with `request.yar.get('lang')`, or `formData.lang` or ..?

Reply:
oki, I can revert back -- this is what I was doing before. I'll just pass in lang to `localisedString` instead